src/lib/Zikula/Composer/Process/RequireJsProcess.php

```diff
@@ -0,0 +1,268 @@
+                    // If the aggregation was successful, add the script to the
+                    // packages array.
+                    $component['main'] = $name.'-built.js';
+
+                    // Add the component to the packages array.
+                    $json['packages'][] = $component;
+                }
+            }
+
+            // Add the shim definition for the package.
+            $shim = isset($options['shim']) ? $options['shim'] : array();
+            if (!empty($shim)) {
+                $json['shim'][$name] = $shim;
+            }
+
+            // Add the config definition for the package.
+            $packageConfig = isset($options['config']) ? $options['config'] : array();
+            if (!empty($packageConfig)) {
+                $json['config'][$name] = $packageConfig;
+            }
+        }
+
+        // Provide the baseUrl.
+        $json['baseUrl'] = $this->baseUrl;
+
+        // Merge in configuration options from the root.
+        if ($this->config->has('component')) {
+            $config = $this->config->get('component');
+            if (isset($config) && is_array($config)) {
+                // Use a recursive, distict array merge.
+                $json = $this->arrayMergeRecursiveDistinct($json, $config);
+            }
+        }
+
+        return $json;
+    }
+
+    /**
+     * Concatenate all scripts together into one destination file.
+     *
+     * @param array $package
+     * @param array $scripts
+     * @param string $file
+     * @return bool
+     */
+    public function aggregateScripts($package, array $scripts, $file)
+    {
+        $assets = $this->newAssetCollection();
+
+        foreach ($scripts as $script) {
+            // Collect each candidate from a glob file search.
+            $path = $this->getVendorDir($package).DIRECTORY_SEPARATOR.$script;
+            $matches = $this->fs->recursiveGlobFiles($path);
+            foreach ($matches as $match) {
+                $assets->add(new FileAsset($match));
+            }
+        }
+        $js = $assets->dump();
+
+        // Write the file if there are any JavaScript assets.
+        if (!empty($js)) {
+            $destination = $this->componentDir.DIRECTORY_SEPARATOR.$file;
+            $this->fs->ensureDirectoryExists(dirname($destination));
+
+            return file_put_contents($destination, $js);
+        }
+
+        return false;
+    }
+
+    /**
+     * Constructs the require.js file from the provided require.js JSON array.
+     *
+     * @param $json
+     *   The require.js JSON configuration.
+     *
+     * @return string
+     *   The RequireJS JavaScript configuration.
+     */
+    public function requireJs(array $json = array())
+    {
+        // Encode the array to a JSON array.
+        $js = JsonFile::encode($json);
+
+        // Construct the JavaScript output.
+        $output = <<<EOT
+var components = $js;
+components.baseUrl = Zikula.Config.baseURL + "web";                
```

Comment:
It will fail of course. https://github.com/zikula/core/blob/7b1c8fdeaade4cf5e85c0843425351314712e163/src/javascript/zikula/zikula.boot.js#L9